common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -537,7 +540,9 @@ void closeAndDeleteOutdatedPartitions(
     partitions
       .forEach((partitionId, partitionInfo) -> {
         synchronized (partitionInfo) {
-          partitionInfo.closeAllFilesAndDeleteIfNeeded(true);
+          AppShufflePartitionInfo.closeAllFilesAndDeleteIfNeeded(true, partitionInfo.dataFile,
```

Comment:
Perhaps we could split `closeAllFilesAndDeleteIfNeeded` into two methods `closeAllFiles` and `deleteAllFiles`, then `closeAllFiles` can be moved into `ResourceCleaner.run()`. 

This way, they don't have to be static. Here, we can change it to call 
```
partitionInfo.cleanable.clean(); 
partitionInfo.deleteAllFiles();
```
and other places only need to call `partitionInfo.cleanable.clean()`.

Reply:
Thanks, modified based on your comments.